cli/cmd/install-cni-plugin.go

```diff
@@ -0,0 +1,166 @@
+package cmd
+
+import (
+	"bytes"
+	"fmt"
+	"io"
+	"os"
+	"strings"
+	"text/template"
+
+	"github.com/linkerd/linkerd2/cli/install"
+	"github.com/linkerd/linkerd2/pkg/version"
+	"github.com/spf13/cobra"
+)
+
+type installCNIPluginConfig struct {
+	Namespace           string
+	CNIPluginImage      string
+	LogLevel            string
+	InboundPort         uint
+	OutboundPort        uint
+	IgnoreInboundPorts  string
+	IgnoreOutboundPorts string
+	ProxyUID            int64
+}
+
+type cniPluginOptions struct {
+	linkerdVersion      string
+	dockerRegistry      string
+	proxyControlPort    uint
+	proxyMetricsPort    uint
+	inboundPort         uint
+	outboundPort        uint
+	ignoreInboundPorts  []uint
+	ignoreOutboundPorts []uint
+	proxyUID            int64
+	cniPluginImage      string
+	logLevel            string
+}
+
+func newCNIPluginOptions() *cniPluginOptions {
+	return &cniPluginOptions{
+		linkerdVersion:      version.Version,
+		dockerRegistry:      defaultDockerRegistry,
+		proxyControlPort:    4190,
+		proxyMetricsPort:    4191,
+		inboundPort:         4143,
+		outboundPort:        4140,
+		ignoreInboundPorts:  nil,
+		ignoreOutboundPorts: nil,
+		proxyUID:            2102,
+		cniPluginImage:      defaultDockerRegistry + "/cni-plugin",
+		logLevel:            "info",
+	}
+}
+
+func (options *cniPluginOptions) validate() error {
+	if !alphaNumDashDot.MatchString(options.linkerdVersion) {
+		return fmt.Errorf("%s is not a valid version", options.linkerdVersion)
+	}
+
+	if !alphaNumDashDotSlashColon.MatchString(options.dockerRegistry) {
+		return fmt.Errorf("%s is not a valid Docker registry. The url can contain only letters, numbers, dash, dot, slash and colon", options.dockerRegistry)
+	}
+
+	found := false
+	allowedLogLevels := [...]string{"trace", "debug", "info", "warn", "error", "fatal", "panic"}
+	for _, level := range allowedLogLevels {
+		if level == options.logLevel {
+			found = true
+		}
+	}
+	if !found {
+		fmt.Printf("%s is not a valid log level for the linkerd-cni plugin so defaulting to warn. Must be one of [trace, debug, info, warn, error, fatal, panic]", options.dockerRegistry)
```

Comment:
logrus provides a `ParseLevel` func that you can use to validate the log level param. We do this in install.go, so I'd use the same approach here.

```diff
diff --git a/cli/cmd/install-cni-plugin.go b/cli/cmd/install-cni-plugin.go
index ba946679..5db12492 100644
--- a/cli/cmd/install-cni-plugin.go
+++ b/cli/cmd/install-cni-plugin.go
@@ -10,6 +10,7 @@ import (
 
 	"github.com/linkerd/linkerd2/cli/install"
 	"github.com/linkerd/linkerd2/pkg/version"
+	log "github.com/sirupsen/logrus"
 	"github.com/spf13/cobra"
 )
 
@@ -63,15 +64,8 @@ func (options *cniPluginOptions) validate() error {
 		return fmt.Errorf("%s is not a valid Docker registry. The url can contain only letters, numbers, dash, dot, slash and colon", options.dockerRegistry)
 	}
 
-	found := false
-	allowedLogLevels := [...]string{"trace", "debug", "info", "warn", "error", "fatal", "panic"}
-	for _, level := range allowedLogLevels {
-		if level == options.logLevel {
-			found = true
-		}
-	}
-	if !found {
-		fmt.Printf("%s is not a valid log level for the linkerd-cni plugin so defaulting to warn. Must be one of [trace, debug, info, warn, error, fatal, panic]", options.dockerRegistry)
+	if _, err := log.ParseLevel(options.logLevel); err != nil {
+		return fmt.Errorf("--cni-log-level must be one of: panic, fatal, error, warn, info, debug")
 	}
 
 	return nil
```